src/GovUk.Education.ExploreEducationStatistics.Common/Extensions/HttpContentExtensions.cs

```diff
@@ -0,0 +1,120 @@
+    /// </para>
+    /// <para>
+    /// Ideally, we'll want to remove this extension if we ever migrate to <see cref="System.Text.Json"/>
+    /// for our API requests and responses.
+    /// </para>
+    /// </remarks>
+    ///
+    /// <param name="content">The JSON serialised HTTP content.</param>
+    /// <param name="sourceEncoding">The content's character encoding.</param>
+    /// <param name="settings">The settings for deserializing the JSON.</param>
+    /// <param name="cancellationToken">A token to signal that the read should be cancelled.</param>
+    /// <typeparam name="T">The type of the object that the JSON should deserialize to.</typeparam>
+    /// <returns>The deserialized object.</returns>
+    public static async Task<T?> ReadFromJsonAsync<T>(
+        this HttpContent content,
+        Encoding sourceEncoding,
+        JsonSerializerSettings? settings = null,
+        CancellationToken cancellationToken = default)
+    {
+        await using var contentStream =
+            await GetContentStreamAsync(content, sourceEncoding, cancellationToken).ConfigureAwait(false);
+
+        using var streamReader = new StreamReader(contentStream);
+        await using var reader = new JsonTextReader(streamReader);
+
+        //
+        return JsonSerializer.Create(settings).Deserialize<T>(reader);
+    }
+
+    public static T? ReadFromJson<T>(
+        this HttpContent content,
+        JsonSerializerSettings? settings = null,
+        CancellationToken cancellationToken = default)
+    {
+        return content.ReadFromJson<T>(Encoding.UTF8, settings, cancellationToken);
+    }
+
+    /// <summary>
+    /// Read JSON serialised content using <see cref="Newtonsoft.Json"/>.
+    /// </summary>
+    ///
+    /// <remarks>
+    /// Prefer to use asynchronous version of this method, however, it can be acceptable
+    /// to use this in test code.
+    /// </remarks>
+    ///
+    /// <param name="content">The JSON serialised HTTP content.</param>
+    /// <param name="sourceEncoding">The content's character encoding.</param>
+    /// <param name="settings">The settings for deserializing the JSON.</param>
+    /// <param name="cancellationToken">A token to signal that the read should be cancelled.</param>
+    /// <typeparam name="T">The type of the object that the JSON should deserialize to.</typeparam>
+    /// <returns>The deserialized object.</returns>
+    public static T? ReadFromJson<T>(
+        this HttpContent content,
+        Encoding sourceEncoding,
+        JsonSerializerSettings? settings = null,
+        CancellationToken cancellationToken = default)
+    {
+        using var contentStream = GetContentStream(content, sourceEncoding, cancellationToken);
+        using var streamReader = new StreamReader(contentStream);
+        using var reader = new JsonTextReader(streamReader);
+
+        return JsonSerializer.Create(settings).Deserialize<T>(reader);
+    }
```

Comment:
Cool, I'm guessing this  to avoid reading the entire string from `HttpContent` first and then using `JsonConvert.DeserializeObject<T>` which would be less optimal?

Reply:
I've added this synchronous version purely for convenience to avoid having to use async/await in the response assertions where we actually assert on the text content in some of the overloads (e.g. `AssertOk`).

Otherwise, we'd end up having to do:

```cs
await response.AssertOk(someViewModel);
```

This seemed a bit weird as assertions typically aren't async!